backend/hitas/models/company.py

```diff
@@ -0,0 +1,137 @@
+
+    if match is None:
+        raise ValidationError(
+            _("%(value)s is not an valid property id"),
+            params={"value": value},
+        )
+
+
+class CompanyState(Enum):
+    NOT_READY = 0
+    LESS_THAN_30_YEARS = 1
+    GREATER_THAN_30_YEARS_NOT_FREE = 2
+    GREATER_THAN_30_YEARS_FREE = 3
+    GREATER_THAN_30_YEARS_PLOT_DEPARTMENT_NOTIFICATION = 4
+    READY_NO_STATISTICS = 5
+
+    class Labels:
+        NOT_READY = _("Not Ready")
+        LESS_THAN_30_YEARS = _("Less than 30 years")
+        GREATER_THAN_30_YEARS_NOT_FREE = _("Greater than 30 years, not free")
+        GREATER_THAN_30_YEARS_FREE = _("Greater than 30 years, free")
+        GREATER_THAN_30_YEARS_PLOT_DEPARTMENT_NOTIFICATION = _("Greater than 30 years, plot department notification")
+        READY_NO_STATISTICS = _("Ready, no statistics")
+
+
+class Company(ExternalHitasModel):
+    # Official spelling of the company name
+    official_name = models.CharField(max_length=1024)
+    # More human-friendly company name
+    display_name = models.CharField(max_length=1024)
+
+    state = EnumIntegerField(CompanyState, default=CompanyState.NOT_READY)
+    # Business ID / 'y-tunnus'
+    business_id = models.CharField(max_length=9, validators=[validate_business_id])
+
+    building_type = models.ForeignKey(BuildingType, on_delete=models.PROTECT)
+    financing_type = models.ForeignKey(FinancingMethod, on_delete=models.PROTECT)
+    public_officer = models.ForeignKey(PublicOfficer, on_delete=models.PROTECT)
+    property_manager = models.ForeignKey(PropertyManager, on_delete=models.PROTECT)
+    developer = models.ForeignKey(Developer, on_delete=models.PROTECT)
+
+    # 'hankinta-arvo'
+    acquisition_price = models.DecimalField(max_digits=15, decimal_places=2)
+    # 'toteutunut hankinta-arvo'
+    realized_acquisition_price = models.DecimalField(null=True, blank=True, max_digits=15, decimal_places=2)
+
+    # 'kiinteistötunnus'
+    property_identifier = models.CharField(max_length=19, validators=[validate_property_id])
+
+    legacy_id = models.CharField(max_length=10, null=True, blank=True)
+
+    class Meta:
+        verbose_name = _("Company")
+        verbose_name_plural = _("Companies")
+
+        constraints = [
+            models.CheckConstraint(name="acquisition_price_positive", check=models.Q(acquisition_price__gte=0)),
+            models.CheckConstraint(
+                name="realized_acquisition_price_positive", check=models.Q(realized_acquisition_price__gte=0)
+            ),
+        ]
+
+    def __str__(self):
+        return self.display_name
+
+
+# Rakennustunnus
+class BuildingIdentifier(HitasModel):
+    value = models.CharField(max_length=1024)
+    company = models.ForeignKey(Company, on_delete=models.PROTECT)
+
+    class Meta:
+        verbose_name = _("Building identifier")
+        verbose_name_plural = _("Building identifiers")
+
+    def __str__(self):
+        return self.value
+
+
+class CompanyAddress(HitasModel):
```

Comment:
I'd rather have a single general Address model, which contains all necessary fields, which Companies, Persons etc. relate to

Reply:
currently I opted not to have this. seems simpler as the address models are not shared (they differ) and there's no need for multiple addresses as real estates and buildings have been added